metadata/component.go

```diff
@@ -94,3 +94,87 @@ func (m *manager) CreateComponent(name string, text string, params param.Params,
 	log.Debugf("Writing component parameters at '%s/%s", componentsDir, name)
 	return m.writeComponentParams(name, params)
 }
+
+func (m *manager) DeleteComponent(name string) error {
+	componentPath, err := m.findComponentPath(name)
+	if err != nil {
+		return err
+	}
+
+	// Delete the component file in components/
+	log.Infof("Deleting component '%s' at path '%s'", name, componentPath)
+	m.appFS.Remove(componentPath)
+
+	// Delete the component references.
+	log.Infof("Removing component parameter references ...")
+
+	// Part of removing the component references imply removing the references
+	// in component/params.libsonnet.
+	log.Debugf("... deleting references in %s", m.componentParamsPath)
+	componentParamsFile, err := afero.ReadFile(m.appFS, string(m.componentParamsPath))
+	if err != nil {
+		return err
+	}
+	jsonnet, err := param.DeleteComponent(name, string(componentParamsFile))
+	if err != nil {
+		return err
+	}
+	err = afero.WriteFile(m.appFS, string(m.componentParamsPath), []byte(jsonnet), defaultFilePermissions)
+	if err != nil {
+		return err
+	}
+
+	// Removing the component references also imply removing the component
+	// references in each environment's environment/<env>/params.libsonnet.
+	envs, err := m.GetEnvironments()
+	if err != nil {
+		return err
+	}
+	for _, env := range envs {
```

Comment:
Ok, my understanding is that we're not actually removing all references to the component, right? Just the entry in the env for the component.

I think that's fine, but we should adjust documentation to note this caveat, and we should also probably make a TODO to call `lint` after we're finished so that we can tell people when they're breaking something.

Reply:
We're removing:
1. The `component/<foo>.jsonnet` file.
2. The references in the `components/params.libsonnet` file.
3. The references in the `environments/<env>/params.libsonnet` file.

What we're not removing and I'll make TODOs for is:
1. References in `main.jsonnet` (mainly because the `main.jsonnet` story is not well scoped and may be subject to change)
2. Component references in other component files, which don't yet exist.